File: mobile/lib/models/mission.dart
class MissionSummary {
  final int id;
  final String name;
  final String? description;
  final String status;
  final DateTime? startDate;
  final DateTime? endDate;
  final double? latitude;
  final double? longitude;
  final double? radiusMeters;
  final List<MissionFormRef> forms;

  MissionSummary({
    required this.id,
    required this.name,
    this.description,
    required this.status,
    this.startDate,
    this.endDate,
    this.latitude,
    this.longitude,
    this.radiusMeters,
    required this.forms,
  });

  factory MissionSummary.fromJson(Map<String, dynamic> json) {
    return MissionSummary(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      status: json['status'] as String,
      startDate: json['startDate'] != null ? DateTime.parse(json['startDate'] as String) : null,
      endDate: json['endDate'] != null ? DateTime.parse(json['endDate'] as String) : null,
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      radiusMeters: (json['radiusMeters'] as num?)?.toDouble(),
      forms: (json['forms'] as List? ?? [])
          .map((f) => MissionFormRef.fromJson(f as Map<String, dynamic>))
          .toList(),
    );
  }
}

class MissionFormRef {
  final int id;
  final String name;

  MissionFormRef({required this.id, required this.name});

  factory MissionFormRef.fromJson(Map<String, dynamic> json) {
    return MissionFormRef(id: json['id'] as int, name: json['name'] as String);
  }
}

File: backend/src/main/java/com/collectpro/backend/entity/Mission.java
package com.collectpro.backend.entity;

import com.collectpro.backend.enums.MissionStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "missions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Mission {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id", nullable = false)
    private Organization organization;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MissionStatus status;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    private Double latitude;

    private Double longitude;

    @Column(name = "radius_meters")
    private Double radiusMeters;

    @Column(name = "expected_collectes_count")
    private Integer expectedCollectesCount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "mission_forms",
            joinColumns = @JoinColumn(name = "mission_id"),
            inverseJoinColumns = @JoinColumn(name = "form_id")
    )
    @Builder.Default
    private Set<Form> forms = new HashSet<>();

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
        if (this.status == null) {
            this.status = MissionStatus.DRAFT;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}